pilopt/src/lib.rs

```diff
@@ -556,6 +556,8 @@ fn remove_trivial_identities<T: FieldElement>(pil_file: &mut Analyzed<T>) {
                 left.expressions.is_empty().then_some(index)
             }
             Identity::Connect(..) => None,
+            // TODO: This is the analog of a permutation, but I don't think this is correct? When is this actually needed?
+            Identity::PhantomBusInteraction(id) => id.tuple.0.is_empty().then_some(index),
```

Comment:
```rust
machine Main with degree: 8 {
    Empty e;
   
    link if 1 ~> e.empty();
}

machine Empty with degree: 8, latch: latch, call_selectors: call_selectors {
    operation empty;

    col fixed latch = [1]*;
}
```

Which right now outputs when optimized:
```rust
namespace main_e(8);
    col witness call_selectors[1];
    main_e::call_selectors[0] * (1 - main_e::call_selectors[0]) = 0;
```

This requires having a single operation in the target, otherwise it would be in the tuple.

Reply:
OK! Sounds like this would not happen in practice. I changed it that we panic in that case, so that we can resolve it if we ever hit this case in practice.